src/lib/src/image/image-loader.component.ts

```diff
@@ -1,60 +1,54 @@
-import { Component, Input, ElementRef, OnChanges, SimpleChanges } from '@angular/core';
+import { Component, Input, OnChanges, SimpleChanges, HostBinding } from '@angular/core';
 import { HalDoc } from '../hal/doc/haldoc';
 
-const PLACEHOLDER = 'url("data:image/gif;base64,R0lGODlhAQABAAAAACwAAAAAAQABAAA=")';
-
 @Component({
   moduleId: module.id,
   selector: 'prx-image',
   template: `
     <img *ngIf="src" [src]="src" (load)="onLoad()" (error)="onError()"/>
+    <img *ngIf="docSrc" [src]="docSrc" (load)="onLoad()" (error)="onError()"/>
   `,
   styleUrls: ['./image-loader.component.css']
 })
 
 export class ImageLoaderComponent implements OnChanges {
   @Input() public src: string;
   @Input() public imageDoc: HalDoc;
+  public docSrc: string;
 
-  constructor(private element: ElementRef) {}
+  @HostBinding('style.background-image') background: string;
+  @HostBinding('class.placeholder') isPlaceholder = false;
+  @HostBinding('class.placeholder-error') isError = false;
 
-  setBackgroundImage(src: string) {
-    if (src) {
-      this.element.nativeElement.style['background-image'] = `url(${src})`;
-    } else {
-      this.element.nativeElement.style['background-image'] = PLACEHOLDER;
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.src) {
+      this.reset();
+    } else if (changes.imageDoc) {
```

Comment:
We're not ever trying to use the `src` and the `imageDoc` at the same time are we? I found a case where both images are displayed but nothing is shown. In the demo, select "Bar Url" src and then select "Doc with image" Image Doc.